appscale/tools/agents/ec2_agent.py

```diff
@@ -191,43 +213,70 @@ def create_security_group(self, parameters, group):
       "name {0}".format(group))
 
 
-  def authorize_security_group(self, parameters, group, from_port, to_port,
-    ip_protocol, cidr_ip):
+  def get_security_group_by_name(self, conn, group):
+    """Gets a security group in AWS with the given name.
```

Comment:
Security group names are unique for classic networking or for a given vpc. To find a group by name you would need to check that the vpc matched (or that the group was not a vpc group)

Reply:
So I need to double check that the vpc ids match?

Reply:
My understanding is that we do not currently check the vpc for the security group during look up and only check that the group name matches.

If looking for a group in a vpc you would check the vpc id of the group matched the configured vpc for appscale. If looking for a classic networking security group (i.e. no configured vpc) you would check that the group was not in a vpc (had no vpc id)